plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/command/valueproviders/MachineHostNameProvider.java

```diff
@@ -32,6 +57,59 @@ public String getKey() {
 
     @Override
     public String getValue() {
-        return Window.Location.getHostName();
+        return value;
+    }
+
+    @Override
+    public void onMachineRunning(MachineStateEvent event) {
+        final MachineStateDto machineState = event.getMachineState();
+        if (machineState.isDev()) {
+            machineServiceClient.getMachine(machineState.getId()).then(new Operation<MachineDto>() {
+                @Override
+                public void apply(MachineDto machine) throws OperationException {
+                    value = getHostName(machine);
+                }
+            });
+        }
+    }
+
+    @Override
+    public void onMachineDestroyed(MachineStateEvent event) {
+        final MachineStateDto machineState = event.getMachineState();
+        if (machineState.isDev()) {
+            value = "";
+        }
+    }
+
+    private void updateValue() {
+        final String devMachineId = appContext.getDevMachineId();
+        if (devMachineId == null) {
+            return;
+        }
+
+        machineServiceClient.getMachine(devMachineId).then(new Operation<MachineDto>() {
+            @Override
+            public void apply(MachineDto machine) throws OperationException {
+                value = getHostName(machine);
+            }
+        }).catchError(new Operation<PromiseError>() {
+            @Override
+            public void apply(PromiseError arg) throws OperationException {
+                value = "";
+            }
+        });
+    }
+
+    private String getHostName(MachineDto machine) {
+        for (ServerDto server : machine.getMetadata().getServers().values()) {
```

Comment:
Is it possible that this method will be called when machine is in state STARTING?